homeassistant/helpers/intent.py

```diff
@@ -110,7 +134,26 @@ def async_validate_slots(self, slots):
     @asyncio.coroutine
     def async_handle(self, intent_obj):
         """Handle the intent."""
-        raise NotImplementedError()
+        if not self.service:
```

Comment:
I don't like this, we're conflating concerns. Please extract this into a new class that extends from `IntentHandler`:

```python
class ServiceIntentHandler(IntentHandler):
```

Reply:
Alternative, instead of a class, you can make it a method.

```python
@bind_hass
@callback
def expose_entity_service_intent(intent_name, domain, service):
```

The automatic wrapping of a service in an intent has a very narrow use case. It only applies to services that only take in an entity_id. If it takes in anything more (like color in light.turn_on), you want to write your own.